test/regexmanager.cpp

```diff
@@ -106,3 +106,33 @@ TEST_CASE("`highlight all` adds rules for all locations", "[RegexManager]")
 		}
 	}
 }
+
+TEST_CASE("RegexManager does not hang on regexes that can match empty strings", "[RegexManager]")
+{
+	RegexManager rxman;
+	std::string input = "The quick brown fox jumps over the lazy dog";
+	std::string compare = input;
+
+	rxman.handle_action("highlight", {"feedlist", "w*", "blue", "red"});
+	rxman.quote_and_highlight(input, "feedlist");
+	REQUIRE(input == "The quick bro<0>w</>n fox jumps over the lazy dog");
+}
+
+TEST_CASE("RegexManager does not hang on regexes that match empty strings", "[RegexManager]")
+{
+	RegexManager rxman;
+	std::string input = "The quick brown fox jumps over the lazy dog";
+	std::string compare = input;
```

Comment:
Please make this `const`, to better convey the meaning.

Reply:
Thank you, I'm never on the outlook for `const`. I should be.